tests/exchange/test_exchange.py

```diff
@@ -378,7 +381,12 @@ def test_get_min_pair_stake_amount(mocker, default_conf) -> None:
         PropertyMock(return_value=markets)
     )
     result = exchange.get_min_pair_stake_amount('ETH/BTC', 1, stoploss)
-    assert isclose(result, 2 * (1+0.05) / (1-abs(stoploss)))
+    expected_result = 2 * (1+0.05) / (1-abs(stoploss))
+    assert isclose(result, expected_result)
+    # With Leverage
+    result = exchange.get_min_pair_stake_amount('ETH/BTC', 1, stoploss, 3.0)
+    assert isclose(result, expected_result/3)
+    # TODO-lev: Min stake for base, kraken and ftx
```

Comment:
just to get this right - are we sure binance and kraken/ftx work differently?

assuming i wanna buy coins for 1000$ with 2x leverage (i input this to the website).

i'll need to have an initial investment (stake amount) of 500$ - i'll not get 2000$ on either of the 3 pages ... or am i missing something?

Now in binance's case, i'll need to first issue a "borrow 500$" call ... but other than that, i'm not sure if they are "really" different (the other 2 basically do the "borrow" automatically) - i'll still get coins worth 1000$.

--- 

Maybe it's also just a question of definition (basically we need to decide how it's going to work).

Assuming a strategy which get's called with returns 1000 ($) from `custom_stake_amount` - and `2` from `leverage()` (still to name/create). 
we can expect the bot to either
* consume 1000$ in stake currency (so your position will be 2000$)
* enter a position of 1000$ - but only using 500$ of your position, borrowing the remaining 500$.

I think i'm fine with either of the two - but it should work the same for all exchanges - which will allow us to set goal on how it's going to work.

Reply:
I'm not completely certain about how the min stake works on kraken/ftx

I've tried to look it up but I can't find anything that specifically addresses leverage. It kinda seemed like the smallest issue to deal with so I've kinda brushed it asside. I can remove the tests if you want, or comment them out

Reply:
it's not about the test (i just happened to think most about it when i looked at this test) - but about a concept - so whenever i switch exchange (with the same strategy / configuration) - i still get the "same outcome" (one of the 2 options above) - not a position twice the size i wanted because we interpret stake_amount for kraken with leverage differently to binance.

Reply:
Yeah I think I'll just test this out on each exchange,
i don't understand what the min is for shorts though based on the data from markets

```
'limits': {
    'amount': {
        'max': 900000.0,
        'min': 1.0
    },
    'cost': {
        'max': None, 
        'min': 10.0
     },
    'market': {
        'max': 376486.87132036, 
        'min': 0.0
    },
    'price': {
        'max': 1000.0, 
        'min': 0.0001
    }
}
```

Reply:
It looks like this is how it is on kraken, I couldn't figure out how to get money into my FTX account

Reply:
> it's not about the test (i just happened to think most about it when i looked at this test) - but about a concept - so whenever i switch exchange (with the same strategy / configuration) - i still get the "same outcome" (one of the 2 options above) - not a position twice the size i wanted because we interpret stake_amount for kraken with leverage differently to binance.

To my understanding, the min_stake_amount is just the minimum amount that you have to put forward for the exchange to accept the trade, so I don't see how this method would affect that, unless that isn't what this method does

Reply:
that's exactly what it does ... 
but as said, don't look at it from the location i put the comment on - but more on a general scale with the sample above.

Assuming a strategy which get's called with returns 1000 ($) from custom_stake_amount - and 2 from leverage() (still to name/create).
we can expect the bot to either

* consume 1000$ in stake currency (so your position will be 2000$)
* enter a position of 1000$ - but only using 500$ of your position, borrowing the remaining 500$.

At the moment, i'm not clear how leverage will be applied ...

Reply:
ok, that's not really a problem for this method though, because this method just tells you what the min amount is

Reply:
I'll adjust set_leverage so it does all this, for binance it'll depend if it's futures or margin also, for margin it will have to borrow

Reply:
> ok, that's not really a problem for this method though, because this method just tells you what the min amount is

that's my point all along - don't look at the LOCATION of the comment / thread :laughing: 
It's a decision we (you?) need to take how it should work - and then we should (try) to implement it consistently across exchanges.

Reply:
what I said was incorrect actually, I think the only difference is that on binance margin(not futures) you need to borrow before placing an order

Reply:
In my opinion, both "set_leverage" and "borrow" must be called from within `create_order` (if necessary).
Especially the "borrow" is binance specific - and the leverage is needed in create_order anyway - as you'll need to pass it as parameter to kraken.


Funnily, the question above (what does stake-amount + leverage mean for the position on exchange) is still open ...

Reply:
stake * lev * price = amount of currency

You don't set the stake amount on the exchange, you set the leverage, and you tell the exchange how much you want, then the exchange determines the stake amount, which will be amount/(price * lev)

Reply:
This is how it works approximately

**Binance**

Request 1 (api/setLeverage)
```
    {
        'leverage': 3.0,
        'pair': 'ADA/USDT'
    }
```
Response
```
    {
        responseCode: 200
    }
```
    
Request 2 (api/order)
```
    {
        amount: 210,
        pair: 'ADA/USDT',
        price: 1
    }
```
    
Response 
```
    {
         code: 200,
         cost: 70
    }
```
    
**Kraken**
Request(api/order)
```
    {
        amount: 210,
        pair: 'ADA/USDT',
        price: 1,
        params: {
            'leverage': 3
        }
    }
```
    
Response 
```
    {
         code: 200,
         cost: 70
    }
```

Reply:
> stake * lev * price = amount of currency
>
>You don't set the stake amount on the exchange, you set the leverage, and you tell the exchange how >much you want, then the exchange determines the stake amount, which will be amount/(price * lev)

Unfortunately that's not what i asked - this is the technical specification of what you send to the exchange (we'll need this too - but at a later point) but doesn't include whatsoever what the meaning of `stake_amount` (either through config or strategy) is in context of leveraged trades.
 
As said above - assuming i set "`take_amount: 1000` (either thorugh config or strategy - that's not really important) - and leverage of 2 - would i expect to get coins worth 2000$ - or coins worth 1000$, using 500$ of the available capital?
It's a very simple question - but it seems like you don't have an answer to it ... ?

Reply:
using $500 and leverage of 2 will give you coins worth $1000, and your stake amount would be $500

Reply:
That's kindof the way i expected it to work (but didn't want to force it that way if the other way would feel more natural.

Unless you need this "thread" for the requests above (that's exactly how expected that to work) - we can mark this as resolved.